### PR TITLE
Add responsive resizing of charts

### PR DESCRIPTION
**Pull request by zachmargolis**
Super basic resizing of the table. It relies on preact to collect duplicate events, so maybe there is some additional debouncing we could add to simplify extra renders if we wanted... (or if we did the SVG chart rendering more directly, we could manage the elements do they wouldn't have to be torn down each time... it's fast enough for now and not causing any problems)

~**Next up**: making the table scrollable, it gets cut off when the breakpoints get too small~ (done, 3918f5a15dd77b4dd5b756aece92e1ed6e9c9a49)

| before | after |
| --- | --- |
| Screen Shot 2021-09-15 at 5 09 28 PM | Screen Shot 2021-09-15 at 5 09 44 PM |
| | Screen Shot 2021-09-15 at 5 10 01 PM |
| | Screen Shot 2021-09-15 at 5 10 09 PM |

### CODE REVIEW

**Comment by aduth** [comment]
Since it's just rendering an `<svg>`, could we tell the plotting library to use percentage-based width instead? `resize` events are pretty performance-intensive.

Looking at the [docs](https://www.npmjs.com/package/@observablehq/plot), maybe:

```js
Plot.plot({
  // ...
  style: {
    width: '80%',
    height: 'auto',
    // max-width: '...'
  },
});
```

**Review by aduth — APPROVED** [review]

File: src/daily-auths-report.tsx [hunk 1]
```diff
@@ -256,16 +262,25 @@ function DailyAuthsReport(): VNode {
...
+    listener();
+
+    window.addEventListener("resize", listener);
+    return () => window.removeEventListener("resize", listener);
+  });
+
   return (
-    <div>
+    <div ref={ref}>
       <PlotComponent
```
Comment (aduth) [inline src/daily-auths-report.tsx:276]: I hadn't noticed 'til now, but it'd be nice to split this file up. Typically I'd only expect one component per file, so pulling out `Plot` to its own file might be a nice way to handle that?

**Comment by zachmargolis** [comment]
> Since it's just rendering an `<svg>`, could we tell the plotting library to use percentage-based width instead? `resize` events are pretty performance-intensive.

Unfortunately, d3 underneath converts everything to pixel space... using `width: 100%` gets gnarly errors like this:

```
attr.js:17 Error: <svg> attribute viewBox: Expected number, "0 0 100% 1800".
```

File: src/daily-auths-report.tsx (hunk 1 above)
Comment (zachmargolis) [reply src/daily-auths-report.tsx:276]: I knew you'd comment on that! 😝 My thinking was more along the lines of "if we aren't going to reference it from a second file, it's fine to keep here for now", but I will move PlotComponent out

Comment (zachmargolis) [reply src/daily-auths-report.tsx:253]: 88e2fa3a29aba7fb6011aedcb68a8f33ae17f3d3

**Comment by aduth** [comment]
> Unfortunately, d3 underneath converts everything to pixel space... using `width: 100%` gets gnarly errors like this:

I was thinking using the `style` option, not the `width` option.

**Comment by zachmargolis** [comment]
> I was thinking using the `style` (`style.width`) option, not the `width` option.

Just gave this a shot...does not seem to resize

```diff
 }): HTMLElement {
   return Plot.plot({
     height: facetAgency ? new Set((data || []).map((d) => d.agency)).size * 60 : undefined,
-     width,
+    style: {
+      width: '100%',
+    },
```

**Pull request merged**